### PR TITLE
fix: add italian HateSpeech dataset

### PR DESCRIPTION
**Pull request by rbroc**
## Checklist for adding MMTEB dataset

**Reason for dataset addition**: This is a highly curated hate speech detection dataset, with binary (hateful/non-hateful) labels, as well as fine-grained labels for 25+ hate speech categories (https://huggingface.co/datasets/Paul/hatecheck-italian).
I am currently only using the binary labels, but it would be cool to also test models on the more fine-grained taxonomy (could just subclass the current task, and set the relevant column as label.

There are similar datasets for 9 more languages, released as separate datasets on HF. If this looks like a suitable dataset, I can go on to submit the remaining (through separate PRs, I assume).

**Checklist**
- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] I have considered the size of the dataset and reduced it if it is too big (2048 examples is typically large enough for most tasks)
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [POINTS.md](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) file.

**Questions** 
- The dataset does not have a native training split (it only comes with a test split). I am currently splitting into equally sized (1845 sentences) train/test splits, stratifying by hate speech type, but we could consider alternative options.
- Don't know what happened with `results/intfloat__multilingual-e5-small/model_meta.json` but can revert if needed

### CODE REVIEW

File: docs/mmteb/points.md
Comment (KennethEnevoldsen) [inline docs/mmteb/points.md:31]: I would add +4 here for adding dataset (first non-translated Italian classification). 

The points for running models are intended for running models on the whole benchmark so those will have to be cut. New task is intended for adding new types of task formulations (e.g. classification vs. retrieval). 

Will you also add a point for the review (me/Kennethenevoldsen)

File: mteb/tasks/Classification/ita/ItaHateClassification.py
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/ita/ItaHateClassification.py:18]: Seems like this is a multilingual dataset. Just letting you know that it is possible to add the full dataset using a multilingual task. This would also allow us to cover Dutch and Portuguese.

File: mteb/tasks/Classification/ita/ItaHateClassification.py
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/ita/ItaHateClassification.py:22]: If you make it multilingual this will turn into a mapping:
```suggestion
        eval_langs={"language_split_name_on_hf": ["ita-Latn"], ... }
```

File: mteb/tasks/Classification/ita/ItaHateClassification.py
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/ita/ItaHateClassification.py:26]: I assume social media? If it is not specified I would add a comment

File: mteb/tasks/Classification/ita/ItaHateClassification.py
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/ita/ItaHateClassification.py:29]: Assuming social media, I would write "mixed"

File: mteb/tasks/Classification/ita/ItaHateClassification.py
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/ita/ItaHateClassification.py:31]:
```suggestion
        dialect=[],
```
basically saying I have considered if there are meaningful dialects to annotate and there is none. As opposed to None which is unannotated.

**Review by KennethEnevoldsen — COMMENTED** [review]
Lovely to see you here @rbroc! Everything is looking good. There are a few comments, but generally it looks good!

File: mteb/tasks/Classification/ita/ItaHateClassification.py
Comment (rbroc) [reply mteb/tasks/Classification/ita/ItaHateClassification.py]: it's sentences created from a template by "annotators", added a comment, but we could also consider updating the domains taxonomy?

Comment (KennethEnevoldsen) [reply mteb/tasks/Classification/ita/ItaHateClassification.py]: Yes, that seems appropriate would "Constructed" work?

File: mteb/tasks/Classification/ita/ItaHateClassification.py
Comment (rbroc) [reply mteb/tasks/Classification/ita/ItaHateClassification.py:22]: related to this and the previous comment, while this is originally a multilingual dataset, it is released on HF as a set of monolingual datasets. Should I combine them? I was originally thinking of just adding them as separate monolingual datasets?

**Comment by rbroc** [comment]
thanks @KennethEnevoldsen! I've left a few comments open above -- especially a question about the multilingual nature of the dataset. Happy to double review points since I am explicitly re-requesting :)

File: mteb/tasks/Classification/ita/ItaHateClassification.py
Comment (rbroc) [reply mteb/tasks/Classification/ita/ItaHateClassification.py]: that sounds good -- I'll go ahead and add it to `abstask/TaskMetadata.py` as part of this PR?

**Comment by KennethEnevoldsen** [comment]
> Happy to double review points since I am explicitly re-requesting :)

No problem, let us just keep it at one.

File: mteb/tasks/Classification/ita/ItaHateClassification.py
Comment (KennethEnevoldsen) [reply mteb/tasks/Classification/ita/ItaHateClassification.py]: Yep that would be great

**Comment by rbroc** [comment]
last quick q @KennethEnevoldsen: are my points for this 2 + 4 (2 for new dataset + 4 point bonus) or 4 in total?
if it looks good, i'll go ahead, update points if needed, and merge and work on a separate PR for the multilingual expansion.

**Review by KennethEnevoldsen — APPROVED** [review]
Looks good!

re. points it is 6 (2+4) in total.

**Comment by rbroc** [comment]
workflows require approval, should be ready to merge after that.

**Comment by KennethEnevoldsen** [comment]
> The dataset does not have a native training split (it only comes with a test split). I am currently splitting into equally sized (1845 sentences) train/test splits, stratifying by hate speech type, but we could consider alternative option

Hei Roberta, just saw this comment. I actually would prefer that you do not do the split, but rather just shuffle and downsample from the training. Then also switch the eval_split to "train". This is to avoid someone training on the "train" split assuming that we test on another split (which we don't).

**Comment by rbroc** [comment]
> > The dataset does not have a native training split (it only comes with a test split). I am currently splitting into equally sized (1845 sentences) train/test splits, stratifying by hate speech type, but we could consider alternative option
> 
> Hei Roberta, just saw this comment. I actually would prefer that you do not do the split, but rather just shuffle and downsample from the training. Then also switch the eval_split to "train". This is to avoid someone training on the "train" split assuming that we test on another split (which we don't).

not sure I fully understand. the dataset has no training split, it only comes with a test split. could also have a chat about this IRL if you think that's more productive.

**Comment by KennethEnevoldsen** [comment]
Ahh then this is perfectly fine. Merged into a temporary branch to run linting and then merging.

**Pull request merged**
